### PR TITLE
Interface with RBFOPT

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
Aaaaaah it does not work for black_box because it is not picklable, and it does not work for rbfopt because it has hidden dependencies to bonmin, which are not installed when you install rbfopt. I see no solution. Grumpf.

File: requirements/dev.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 black==20.8b1
+rbfopt>=4.2.2
```
Comment (jrapin) [inline requirements/dev.txt:2]: this does not seem standard, please move this to bench.txt

File: nevergrad/optimization/recastlib.py [hunk 2]
```diff
@@ -59,16 +71,51 @@ def _optimization_function(self, objective_function: tp.Callable[[tp.ArrayLike],
...
+                    np.array([-1.0] * d),
+                    np.array([1.0] * d),
+                    np.array(["R"] * d),
+                    self.local_objective_function,
+                )
+                settings = rbfopt.RbfoptSettings(max_evaluations=budget)
+                alg = rbfopt.RbfoptAlgorithm(settings, bb)
+                val, x, _, _, fast_evalcount = alg.optimize()
+                if val < best_res:
+                    best_x = np.arctanh(x)
```
Comment (jrapin) [inline nevergrad/optimization/recastlib.py:106]: I would avoid tanh, really, this clips too fast. tan/arctan is better

File: nevergrad/optimization/recastlib.py [hunk 3]
```diff
@@ -49,8 +57,12 @@ def get_optimization_function(self) -> tp.Callable[[tp.Callable[[tp.ArrayLike],
         subinstance.current_bests = self.current_bests
         return subinstance._optimization_function
 
+    def local_objective_function(self, x) -> float:
```
Comment (jrapin) [inline nevergrad/optimization/recastlib.py:60]: please make this kind of methods private

**Review by jrapin — CHANGES_REQUESTED** [review]

**Comment by teytaud** [comment]
It's sad but RBFOPT has no proper way to install its dependencies. This can not work unless some apt-get have been done. 
Maybe I could ensure that this works for users who do have the subdependencies installed, but that makes our code more complicated and difficult to maintain. I prefer to close for now.

**Pull request closed without merging**